scripts/container-e2e.sh

```
@@ -0,0 +1,87 @@
+# Requires: docker, curl, and `buf` on PATH (for `buf curl`).
+# Usage: scripts/container-e2e.sh <image> <label>
+set -uo pipefail
+
+IMAGE="$1"; LABEL="$2"
+NAME="e2e-$LABEL-$$"
+PORT="${E2E_PORT:-15000}"
+PASS=0; FAIL=0
+
+ok()   { printf '  \033[32mPASS\033[0m %s\n' "$1"; PASS=$((PASS+1)); }
+bad()  { printf '  \033[31mFAIL\033[0m %s -- %s\n' "$1" "$2"; FAIL=$((FAIL+1)); }
+cleanup() { docker rm -f "$NAME" >/dev/null 2>&1 || true; }
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)